assets/src/dashboard/utils/useGenericDashboardView.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { sprintf, _n } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import { useMemo } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { RESULT_LABELS } from '../constants';
+
+export default function useGenericDashboardView({
```

Comment:
I think we can be more specific with this and call it `useDashboardFilterLabel`

Reply:
Sure! My thought was that it would be the one stop shop for all the generic pieces that get assembled across the views, but having them separate is probably actually more straight forward and built as needed.